common/djangoapps/util/tests/test_request.py

```diff
@@ -50,8 +50,26 @@ def test_course_id_from_url(self):
         self.assertIsNone(course_id_from_url('/login'))
         self.assertIsNone(course_id_from_url('/course/edX/maths/2020'))
         self.assertIsNone(course_id_from_url('/courses/edX/maths/'))
+        self.assertIsNone(course_id_from_url('/api/courses/v1/blocks/edX/maths/2020'))
+        self.assertIsNone(course_id_from_url('/api/courses/v1/blocks/course-v1:incidental+courseid+formatting'))
+        self.assertIsNone(course_id_from_url('/api/courses/v41/notcourses/course-v1:incidental+courseid+formatting'))
 
         course_id = course_id_from_url('/courses/edX/maths/2020')
         self.assertEqual(course_id.org, 'edX')
         self.assertEqual(course_id.course, 'maths')
         self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/courses/course-v1:edX+maths+2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/api/courses/v1/courses/course-v1:edX+maths+2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/api/courses/v1/courses/edX/maths/2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
```

Comment:
I'm going with the former approach. I don't think I'm a fan of the approach of importing/constructing a whole object to only have specific fields and then asserting the equality of the object and the output of what we are testing- what if the implementation of that object's initializer changes elsewhere but doesn't affect the actual functionality being tested here?